phys2bids/phys2bids.py

```diff
@@ -309,71 +220,163 @@ def phys2bids(filename, info=False, indir='.', outdir='.', heur_file=None,
-
-    LGR.info(f'Preparing {output_amount} output files.')
-    phys_out = {}  # create phys_out dict that will have a
-    # blueprint object per frequency
-    # for each different frequency
-    for uniq_freq in uniq_freq_list:
-        # copy the phys_in object to the new dict entry
-        phys_out[uniq_freq] = deepcopy(phys_in)
-        # this counter will take into account how many channels are eliminated
-        count = 0
-        # for each channel in the original phys_in object
-        # take the frequency
-        for idx, i in enumerate(phys_in.freq):
-            # if that frequency is different than the frequency of the phys_obj entry
-            if i != uniq_freq:
-                # eliminate that channel from the dict since we only want channels
-                # with the same frequency
-                phys_out[uniq_freq].delete_at_index(idx - count)
-                # take into acount the elimination so in the next eliminated channel we
-                # eliminate correctly
-                count += 1
-        # Also create a BlueprintOutput object for each unique frequency found.
-        # Populate it with the corresponding blueprint input and replace it
-        # in the dictionary.
-        phys_out[uniq_freq] = BlueprintOutput.init_from_blueprint(phys_out[uniq_freq])
-
-    if heur_file and sub:
-        LGR.info(f'Preparing BIDS output using {heur_file}')
-    elif heur_file and not sub:
-        LGR.warning('While "-heur" was specified, option "-sub" was not.\n'
-                    'Skipping BIDS formatting.')
-
-    # Preparing output parameters: name and folder.
-    for uniq_freq in uniq_freq_list:
-        # If possible, prepare bids renaming.
-        if heur_file and sub:
-            if output_amount > 1:
-                # Add "recording-freq" to filename if more than one freq
-                outfile = use_heuristic(heur_file, sub, ses, filename,
-                                        outdir, uniq_freq)
-            else:
-                outfile = use_heuristic(heur_file, sub, ses, filename, outdir)
+    # objects, one for each unique frequency for each run in phys_in
+    # they also save the amount of runs and unique frequencies
+    uniq_freq_list = set(phys_in[1].freq)
+    freq_amount = len(uniq_freq_list)
+    if freq_amount > 1:
+        LGR.info(f'Found {freq_amount} different frequencies in input!')
+
+    # If heuristics are used, init a dict of arguments to pass to use_heuristic
+    if heur_file is not None and sub is not None:
+        heur_args = {'heur_file': heur_file, 'sub': sub, 'ses': ses,
+                     'filename': filename, 'outdir': outdir, 'run': '',
+                     'record_label': ''}
+
+    LGR.info(f'Preparing {freq_amount*run_amount} output files.')
+    # Create phys_out dict that will have a blueprint object for each different frequency
+    phys_out = {}
+
+    # Export a (set of) phys_out for each element in phys_in
+    # what's the run key ??? 1 or 0
```

Comment:
I believe this is something we already discussed. We should probably update this comment to what was decided.